GitDepend.UnitTests/Visitors/CheckArtifactsVisitorTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO.Abstractions;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GitDepend.Busi;
+using GitDepend.Visitors;
+using NUnit.Framework;
+
+namespace GitDepend.UnitTests.Visitors
+{
+    [TestFixture]
+    public class CheckArtifactsVisitorTests : TestFixtureBase
+    {
+        private IFileSystem _fileSystem;
+        [SetUp]
+        public void Setup()
+        {
+            _fileSystem = RegisterMockFileSystem();
+            EnsureDirectory(_fileSystem, Lib1Directory);
+            EnsureDirectory(_fileSystem, Lib2Directory);
+            EnsureDirectory(_fileSystem, Lib1PackagesDirectory);
+            EnsureDirectory(_fileSystem, Lib2PackagesDirectory);
+        }
+
+        [Test]
+        public void ArtifactsUpToDate_ReturnsSuccess()
+        {
+            EnsureFiles(_fileSystem, Lib1PackagesDirectory, Lib1Packages);
+
+            var visitor = new CheckArtifactsVisitor();
+            var code = visitor.VisitDependency(Lib1Directory, Lib1Dependency);
+
+            Assert.AreEqual(ReturnCode.Success, code);
+        }
+
+        [Test]
+        public void ArtifactsDontExist_ReturnsOutOfDate()
+        {
+            var visitor = new CheckArtifactsVisitor();
+            var code = visitor.VisitDependency(Lib1Directory, Lib1Dependency);
+
+            Assert.AreEqual(ReturnCode.DependencyPackagesNotBuilt, code);
```

Comment:
anything other than `ReturnCode.Success` is designed to exit the program immediately. This appears to be using this value as a flow control within the program, but will not be used as an exit code for the process itself. I would prefer that you add a Boolean property to indicate that everything is up to date and key off of that. Am I understanding this code correctly?